rskj-core/src/test/java/co/rsk/peg/BridgeSupportSigHashTest.java

```diff
@@ -0,0 +1,292 @@
+            .withProvider(provider)
+            .withActivations(activations)
+            .build();
+
+        // Act
+        bridgeSupport.updateCollections(mock(Transaction.class));
+
+        // Assertions
+
+        // Assert one pegout tx was added to pegoutsWaitingForConfirmations from the creation of a pegout batch
+        assertEquals(1, pegoutsWaitingForConfirmations.getEntries().size());
+
+        if (activations.isActive(ConsensusRule.RSKIP379)){
+            PegoutsWaitingForConfirmations.Entry pegoutBatchTx = pegoutsWaitingForConfirmations.getEntries().stream().findFirst().get();
+            Optional<Sha256Hash> firstInputSigHash = BitcoinUtils.getFirstInputSigHash(pegoutBatchTx.getBtcTransaction());
+            assertTrue(firstInputSigHash.isPresent());
+            verify(provider, times(1)).setPegoutTxSigHash(
+                firstInputSigHash.get()
+            );
+        } else {
+            verify(provider, never()).hasPegoutTxSigHash(
+                any()
+            );
+            // verify no sigHash was added to sigHashes list before RSKIP379
+            verify(provider, never()).setPegoutTxSigHash(
+                any()
+            );
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("pegoutTxIndexArgsProvider")
+    void test_pegoutTxIndex_when_migration_tx_is_created(ActivationConfig.ForBlock activations) throws IOException {
+        // Arrange
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        when(provider.getFeePerKb())
+            .thenReturn(Coin.MILLICOIN);
+
+        when(provider.getReleaseRequestQueue())
+            .thenReturn(new ReleaseRequestQueue(Collections.emptyList()));
+
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = new PegoutsWaitingForConfirmations(new HashSet<>());
+        when(provider.getPegoutsWaitingForConfirmations())
+            .thenReturn(pegoutsWaitingForConfirmations);
+
+        when(provider.getPegoutsWaitingForSignatures())
+            .thenReturn(new TreeMap<>());
+
+        Federation oldFederation = bridgeMainnetConstants.getGenesisFederation();
+        Federation newFederation = new Federation(
+            FederationTestUtils.getFederationMembers(1),
+            Instant.EPOCH,
+            5L,
+            btcMainnetParams
+        );
+        when(provider.getOldFederation())
+            .thenReturn(oldFederation);
+        when(provider.getNewFederation())
+            .thenReturn(newFederation);
+
+        // Utxos to migrate
+        List<UTXO> utxos = PegTestUtils.createUTXOs(10, oldFederation.getAddress());
+        when(provider.getOldFederationBtcUTXOs())
+            .thenReturn(utxos);
+
+        // Advance blockchain to migration phase
+        long migrationAge = bridgeMainnetConstants.getFederationActivationAge(activations) +
+                                bridgeMainnetConstants.getFundsMigrationAgeSinceActivationBegin() + 6;
```

Comment:
What Does The 6 Represent?

Reply:
This represents the block activation of the new fed(5) + 1 block. The migration starts 1 block after the migration age is reached.

Reply:
I'm going to move that number to a variable and also I'll add a comment..

Reply:
Done